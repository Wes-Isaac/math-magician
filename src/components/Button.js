import { Component } from 'react';

// eslint-disable-next-line react/prefer-stateless-function
class Button extends Component {
  render() {
    const { buttonName, css, handleClick } = this.props;
    return (
      <button className={css} onClick={() => handleClick(buttonName)}>
        {buttonName}
      </button>
    );
  }
}

export default Button;
