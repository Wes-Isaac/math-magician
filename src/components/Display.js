import { Component } from 'react';

// eslint-disable-next-line react/prefer-stateless-function
class Display extends Component {
  render() {
    const {
      css,
      total,
      next,
      operation,
    } = this.props;
    return (
      <div className={css}>
        {total}
        {operation}
        {next}
      </div>
    );
  }
}

export default Display;
