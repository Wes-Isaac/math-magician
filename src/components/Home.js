import React from 'react'

export default function Home() {
  return (
    <div className="home">
      <h2>Welcome to our page.</h2>
      <p>Lorem ipsum, dolor sit amet consectetur adipisicing elit. Corrupti deserunt omnis culpa possimus at illum recusandae, architecto assumenda autem, repellendus ipsum suscipit alias aperiam fugit! Repellendus asperiores voluptatem quibusdam sapiente nisi error blanditiis, laboriosam obcaecati et dignissimos quod temporibus incidunt, tenetur velit. Debitis reprehenderit deserunt alias, dolore ipsam impedit quasi distinctio, praesentium, explicabo nesciunt sunt qui cum adipisci quas consequuntur.Debitis reprehenderit deserunt alias, dolore ipsam impedit quasi distinctio, praesentium, explicabo nesciunt sunt qui cum adipisci quas consequuntur.</p>
      <p>Lorem ipsum, dolor sit amet consectetur adipisicing elit. Corrupti deserunt omnis culpa possimus at illum recusandae, architecto assumenda autem, repellendus ipsum suscipit alias aperiam fugit! Repellendus asperiores voluptatem quibusdam sapiente nisi error blanditiis, laboriosam obcaecati et dignissimos quod temporibus incidunt, tenetur velit. Debitis reprehenderit deserunt alias, dolore ipsam impedit quasi distinctio, praesentium, explicabo nesciunt sunt qui cum adipisci quas consequuntur.Debitis reprehenderit deserunt alias, dolore ipsam impedit quasi distinctio, praesentium, explicabo nesciunt sunt qui cum adipisci quas consequuntur.</p>
  
    </div>
  )
}
