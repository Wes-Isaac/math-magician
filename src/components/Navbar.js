import React from 'react'
import { Link } from 'react-router-dom'

export default function Navbar() {
  return (
    <nav className="navbar">
      <h1>Math Magician</h1>
      <div className="links">
        <Link to="/">Home</Link>
        <Link to="/calculator">Calculator</Link>
        <Link to="/quote">Quote</Link>
      </div>
    </nav>
  )
}
