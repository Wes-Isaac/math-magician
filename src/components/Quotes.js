import React from 'react';

export default function Quotes() {
  return (
    <div className="quote">
      <p>
        Mathematics is not about numbers, equations, computations, or
        algorithms: it is about understanding. - William Paul Thurston
      </p>
    </div>
  );
}
