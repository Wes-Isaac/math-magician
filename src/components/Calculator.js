import React, { useState } from 'react';
import Button from './Button';
import Display from './Display';
import calculate from '../logic/calculate';
import './calculator.css';

const Calculator = () => {
  const [total, setTotal] = useState(0);
  const [next, setNext] = useState(null);
  const [operation, setOperation] = useState(null);
  const handleClick = (val) => {
    const res = calculate({ total, next, operation }, val);
    setTotal(res.total);
    setNext(res.next);
    setOperation(res.operation);
  };

  return (
    <div className="container">
      <p>Let&apos;s do some Math!</p>
      <div className="calculator-grid">
        <Display css="output" total={total} next={next} operation={operation} />
        <Button buttonName="AC" handleClick={handleClick} />
        <Button buttonName="+/-" handleClick={handleClick} />
        <Button buttonName="%" handleClick={handleClick} />
        <Button buttonName="รท" css="bg-secondary" handleClick={handleClick} />
        <Button buttonName="7" handleClick={handleClick} />
        <Button buttonName="8" handleClick={handleClick} />
        <Button buttonName="9" handleClick={handleClick} />
        <Button buttonName="x" css="bg-secondary" handleClick={handleClick} />
        <Button buttonName="4" handleClick={handleClick} />
        <Button buttonName="5" handleClick={handleClick} />
        <Button buttonName="6" handleClick={handleClick} />
        <Button buttonName="-" css="bg-secondary" handleClick={handleClick} />
        <Button buttonName="1" handleClick={handleClick} />
        <Button buttonName="2" handleClick={handleClick} />
        <Button buttonName="3" handleClick={handleClick} />
        <Button buttonName="+" css="bg-secondary" handleClick={handleClick} />
        <Button buttonName="0" css="span-two" handleClick={handleClick} />
        <Button buttonName="." handleClick={handleClick} />
        <Button buttonName="=" css="bg-secondary" handleClick={handleClick} />
      </div>
    </div>
  );
};
export default Calculator;
