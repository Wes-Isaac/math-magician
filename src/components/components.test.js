import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import Calculator from './Calculator';
import Display from './Display';
import Home from './Home';
import Quotes from './Quotes';
import Button from './Button';
import Navbar from './Navbar';

describe('test screenshots for all components', () => {
  test('matches Display snapshot', () => {
    const tree = renderer.create(<Display />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  test('matches Calculator snapshot', () => {
    const tree = renderer.create(<Calculator />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  test('matches Home snapshot', () => {
    const tree = renderer.create(<Home />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  test('matches Quote snapshot', () => {
    const tree = renderer.create(<Quotes />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  test('matches Button snapshot', () => {
    const tree = renderer.create(<Button />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  test('matches Navbar snapshot', () => {
    const tree = renderer.create(
      <Router>
        <Navbar />
      </Router>,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
