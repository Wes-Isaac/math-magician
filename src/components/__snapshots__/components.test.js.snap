// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test screenshots for all components matches Button snapshot 1`] = `
<button
  onClick={[Function]}
/>
`;

exports[`test screenshots for all components matches Calculator snapshot 1`] = `
<div
  className="container"
>
  <p>
    Let's do some Math!
  </p>
  <div
    className="calculator-grid"
  >
    <div
      className="output"
    >
      0
    </div>
    <button
      onClick={[Function]}
    >
      AC
    </button>
    <button
      onClick={[Function]}
    >
      +/-
    </button>
    <button
      onClick={[Function]}
    >
      %
    </button>
    <button
      className="bg-secondary"
      onClick={[Function]}
    >
      รท
    </button>
    <button
      onClick={[Function]}
    >
      7
    </button>
    <button
      onClick={[Function]}
    >
      8
    </button>
    <button
      onClick={[Function]}
    >
      9
    </button>
    <button
      className="bg-secondary"
      onClick={[Function]}
    >
      x
    </button>
    <button
      onClick={[Function]}
    >
      4
    </button>
    <button
      onClick={[Function]}
    >
      5
    </button>
    <button
      onClick={[Function]}
    >
      6
    </button>
    <button
      className="bg-secondary"
      onClick={[Function]}
    >
      -
    </button>
    <button
      onClick={[Function]}
    >
      1
    </button>
    <button
      onClick={[Function]}
    >
      2
    </button>
    <button
      onClick={[Function]}
    >
      3
    </button>
    <button
      className="bg-secondary"
      onClick={[Function]}
    >
      +
    </button>
    <button
      className="span-two"
      onClick={[Function]}
    >
      0
    </button>
    <button
      onClick={[Function]}
    >
      .
    </button>
    <button
      className="bg-secondary"
      onClick={[Function]}
    >
      =
    </button>
  </div>
</div>
`;

exports[`test screenshots for all components matches Display snapshot 1`] = `<div />`;

exports[`test screenshots for all components matches Home snapshot 1`] = `
<div
  className="home"
>
  <h2>
    Welcome to our page.
  </h2>
  <p>
    Lorem ipsum, dolor sit amet consectetur adipisicing elit. Corrupti deserunt omnis culpa possimus at illum recusandae, architecto assumenda autem, repellendus ipsum suscipit alias aperiam fugit! Repellendus asperiores voluptatem quibusdam sapiente nisi error blanditiis, laboriosam obcaecati et dignissimos quod temporibus incidunt, tenetur velit. Debitis reprehenderit deserunt alias, dolore ipsam impedit quasi distinctio, praesentium, explicabo nesciunt sunt qui cum adipisci quas consequuntur.Debitis reprehenderit deserunt alias, dolore ipsam impedit quasi distinctio, praesentium, explicabo nesciunt sunt qui cum adipisci quas consequuntur.
  </p>
  <p>
    Lorem ipsum, dolor sit amet consectetur adipisicing elit. Corrupti deserunt omnis culpa possimus at illum recusandae, architecto assumenda autem, repellendus ipsum suscipit alias aperiam fugit! Repellendus asperiores voluptatem quibusdam sapiente nisi error blanditiis, laboriosam obcaecati et dignissimos quod temporibus incidunt, tenetur velit. Debitis reprehenderit deserunt alias, dolore ipsam impedit quasi distinctio, praesentium, explicabo nesciunt sunt qui cum adipisci quas consequuntur.Debitis reprehenderit deserunt alias, dolore ipsam impedit quasi distinctio, praesentium, explicabo nesciunt sunt qui cum adipisci quas consequuntur.
  </p>
</div>
`;

exports[`test screenshots for all components matches Navbar snapshot 1`] = `
<nav
  className="navbar"
>
  <h1>
    Math Magician
  </h1>
  <div
    className="links"
  >
    <a
      href="/"
      onClick={[Function]}
    >
      Home
    </a>
    <a
      href="/calculator"
      onClick={[Function]}
    >
      Calculator
    </a>
    <a
      href="/quote"
      onClick={[Function]}
    >
      Quote
    </a>
  </div>
</nav>
`;

exports[`test screenshots for all components matches Quote snapshot 1`] = `
<div
  className="quote"
>
  <p>
    Mathematics is not about numbers, equations, computations, or algorithms: it is about understanding. - William Paul Thurston
  </p>
</div>
`;
